parser/src/lexer.rs

```diff
@@ -1105,6 +1105,84 @@ where
         let tok_end = self.get_pos();
         Ok((tok_start, ty, tok_end))
     }
+
+    fn lex_byte(&self, s: String) -> Result<Vec<u8>, LexicalError> {
```

Comment:
This function could be a function outside of the `Lexer` struct, since self was never referenced. I would prefer to extract this function out of the struct.